hash/hash.go

```diff
@@ -0,0 +1,101 @@
+// Package hash provides utility functions to process complex data types, like
+// data streams, files, or sequences of structs of different types.
+package hash
+
+import (
+	"bytes"
+	"errors"
+	h "hash"
```

Comment:
style: I prefer aliasing packages to other names - makes it harder to read the code.